### PR TITLE
Add hyper_reqwest example

### PR DESCRIPTION
**Pull request by Asakiz**
Signed-off-by: asakiz <asakizin@gmail.com>

### CODE REVIEW

File: actix_full/src/main.rs [hunk 1]
```diff
@@ -121,6 +121,17 @@ impl rwcst::AppImpl for App {
...
+            let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+
+            let make_svc =
+                make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(handle)) });
+
+            Server::bind(&addr).serve(make_svc);
+
+            Ok(())
+        }
+
```
Comment (Jonathas-Conceicao) [inline actix_full/src/main.rs]: We shouldn't have changes to the actix_full example in this PR

File: hyper_reqwest/Cargo.toml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+actix-rt = "1"
+actix-web = { version = "3.0.0-alpha.1", default-features = false }
+async-trait = "0.1"
+derive_more = { version = "0.99", default-features = false, features = ["from"] }
+futures-util = "0.3"
+hyper = "0.13"
+reqwest = { version = "0.10", default-features = false, features = ["json", "native-tls"] }
+rwcst = { path = "..", package = "rust-web-client-server-testing" }
+tokio = { version = "0.2", features = ["macros"] }
+serde_json = "1.0"
```
Comment (Jonathas-Conceicao) [inline hyper_reqwest/Cargo.toml]: new line here

File: actix_full/src/main.rs [hunk 3]
```diff
@@ -121,6 +121,17 @@ impl rwcst::AppImpl for App {
...
+        fn serve(&mut self) -> Result<()> {
+            let addr = SocketAddr::from(([127, 0, 0, 1], 3000));
+
+            let make_svc =
+                make_service_fn(|_conn| async { Ok::<_, Infallible>(service_fn(handle)) });
+
+            Server::bind(&addr).serve(make_svc);
+
+            Ok(())
+        }
```
Comment (otavio) [inline actix_full/src/main.rs]: This should be made in another commit as this is not related.

**Review by otavio — CHANGES_REQUESTED** [review]

File: actix_full/src/main.rs (hunk 3 above)
Comment (Asakiz) [reply actix_full/src/main.rs]: Done!

File: hyper_reqwest/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+                }
+            }
+        }
+        let state = self.info.clone();
+        let make_svc = make_service_fn(move |_conn| {
+            let state = state.clone();
+            async move {
+                Ok::<_, Infallible>(service_fn(move |req: Request<Body>| {
+                    let state = state.clone();
+                    async move { Ok::<_, Infallible>(handle(req, state).await) }
```
Comment (Jonathas-Conceicao) [inline hyper_reqwest/src/main.rs]: The handle function might fail if the route requested is no the expected one, and yet, here is set to a `Result<_, Infallible>`. I think the `handle` function should be reworked to return a `Result`.

**Review by Jonathas-Conceicao — CHANGES_REQUESTED** [review]

File: actix_full/src/main.rs (hunk 1 above)
Comment (Asakiz) [reply actix_full/src/main.rs]: Done!

File: hyper_reqwest/Cargo.toml (hunk 2 above)
Comment (Asakiz) [reply hyper_reqwest/Cargo.toml]: Done!

File: hyper_reqwest/src/main.rs (hunk 4 above)
Comment (Asakiz) [reply hyper_reqwest/src/main.rs]: Done!

File: hyper_reqwest/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+        }
+        let state = self.info.clone();
+        let make_svc = make_service_fn(move |_conn| {
+            let state = state.clone();
+            async move {
+                Ok::<_, Infallible>(service_fn(move |req: Request<Body>| {
+                    let state = state.clone();
+                    async move { Ok::<_, Infallible>(handle(req, state).await.unwrap()) }
+                }))
+            }
```
Comment (Jonathas-Conceicao) [inline hyper_reqwest/src/main.rs]: Should be:
```Rust
            async move {
                Ok::<_, Infallible>(service_fn(move |req: Request<Body>| {
                    let state = state.clone();
                    handle(req, state)
                }))
            }
```

To do so just add the proper derives to the local `Error` type.

**Review by Jonathas-Conceicao — CHANGES_REQUESTED** [review]

File: hyper_reqwest/src/main.rs (hunk 5 above)
Comment (Asakiz) [reply hyper_reqwest/src/main.rs]: Done!

File: hyper_reqwest/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,135 @@
...
+        ) -> Result<Response<Body>> {
+            match (req.method(), req.uri().path()) {
+                (&Method::GET, "/") => {
+                    use std::ops::Deref;
+
+                    let state = state.lock().await;
+                    let body = serde_json::to_string(&state.deref())?;
+                    Ok(Response::new(Body::from(body)))
+                }
+                _ => Ok(Response::builder().status(StatusCode::NOT_FOUND).body(NOTFOUND.into())?),
```
Comment (Jonathas-Conceicao) [inline hyper_reqwest/src/main.rs]: This should be an `Err`, no point in having a `Result` return type if we don't use it.

**Review by Jonathas-Conceicao — CHANGES_REQUESTED** [review]

File: hyper_reqwest/src/main.rs (hunk 6 above)
Comment (Asakiz) [reply hyper_reqwest/src/main.rs]: Done!

**Review by Jonathas-Conceicao — APPROVED** [review]
